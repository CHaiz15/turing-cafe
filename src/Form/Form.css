.res-form {
  widows: 100vw;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.form-btn {
  background-color: #028080;
  color: white;
  width: 50%;
  height: 36px;
  margin: 1%;
}

input {
  height: 30px;
  margin: 1%;
}
