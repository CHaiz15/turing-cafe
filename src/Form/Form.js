import React, { Component } from 'react'
import './Form.css'


class Form extends Component {
  constructor() {
    super();

    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleClick = () => {
    this.setState({name: '', date: '', time: '', number: ''})
    this.props.addReservation(this.state)
  }

  render() {
    return (
      <div className='res-form'>
        <input
          placeholder='Name'
          name='name'
          value={this.state.name}
          onChange={this.handleChange}
        />
        <input 
          placeholder='Date (mm/dd)'
          name='date'
          value={this.state.date}
          onChange={this.handleChange}
        />
        <input 
          placeholder='Time'
          name='time'
          value={this.state.time}
          onChange={this.handleChange}
        />
        <input 
          placeholder='Number of guests'
          name='number'
          value={this.state.number}
          onChange={this.handleChange}
        />
        <button onClick={this.handleClick}>Make Reservation</button>
      </div>
    )
  }
}

export default Form