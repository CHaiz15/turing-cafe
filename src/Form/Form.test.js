import React from 'react';
import Form from './Form';
import { shallow } from 'enzyme'

it('should be an instance of an Form componenet', () => {
  const wrapper = shallow(
    <Form 
      addReservation={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change state when handleChange is called', () => {
  const wrapper = shallow(
    <Form 
      addReservation={jest.fn()}
    />
  );
  const mockEvent = {
    preventDefault: jest.fn(),
    target: {
      name: 'name',
      value: 'Richie'
    }};
  wrapper.instance().handleChange(mockEvent)
  expect(wrapper.state('name')).toEqual('Richie')
})

it('should call addReservation when button is clicked', () => {
  const addReservationMock = jest.fn();
  const wrapper = shallow(
    <Form 
      addReservation={addReservationMock}
    />
  );
  wrapper.find('.form-btn').simulate('click')
  expect(addReservationMock).toHaveBeenCalled();
})

it('should change state when handleClick is called', () => {
  const wrapper = shallow(
    <Form 
      addReservation={jest.fn()}
    />
  );
  const defaultState = {name: 'Jeff', date: '04/23', time: '7:00', number: '16'}
  const expected = {name: '', date: '', time: '', number: ''}
  wrapper.instance().setState(defaultState);
  wrapper.instance().handleClick();
  expect(wrapper.state()).toEqual(expected);
})
