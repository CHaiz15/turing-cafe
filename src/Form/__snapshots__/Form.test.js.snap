// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be an instance of an Form componenet 1`] = `
<div
  className="res-form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    value=""
  />
  <button
    className="form-btn"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</div>
`;
